src/app/util/ClientMonitoringRegistrationTable.cpp

```diff
@@ -17,59 +17,71 @@
 
-void ClientMonitoringRegistrationTable::LoadFromStorage(FabricIndex fabricIndex)
+/**
+ * @brief Function loads a client registration entry from persistent storage for a single fabric
+ *
+ * @param fabricIndex fabric index to load from storage
+ * @return CHIP_ERROR
+ */
+CHIP_ERROR ClientMonitoringRegistrationTable::LoadFromStorage(FabricIndex fabricIndex)
 {
-    // TODO: Implement load from NVM logic
-}
+    uint8_t buffer[kRegStorageSize] = { 0 };
+    uint16_t size                   = sizeof(buffer);
 
-void ClientMonitoringRegistrationTable::SaveToStorage()
-{
-    // Store to NVM based of class attributes
-}
+    ReturnErrorOnFailure(
+        mStorage.SyncGetKeyValue(DefaultStorageKeyAllocator::ClientMonitoringTableEntry(fabricIndex).KeyName(), buffer, size));
 
-NodeId ClientMonitoringRegistrationTable::getClientNodeId()
-{
-    return mRegisteredClient.clientNodeId;
-}
+    TLV::TLVReader reader;
+    reader.Init(buffer, size);
+    ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Structure, TLV::AnonymousTag()));
 
-uint64_t ClientMonitoringRegistrationTable::getICid()
-{
-    return mRegisteredClient.ICid;
-}
+    ReturnErrorOnFailure(mRegisteredClient.Decode(reader));
 
-FabricIndex ClientMonitoringRegistrationTable::getFaricIndex()
-{
-    return mRegisteredClient.fabricIndex;
-}
+    mRegisteredClient.fabricIndex = fabricIndex;
 
-void ClientMonitoringRegistrationTable::setClientNodeId(NodeId clientNodeId)
-{
-    mRegisteredClient.clientNodeId = clientNodeId;
+    return CHIP_NO_ERROR;
 }
 
-void ClientMonitoringRegistrationTable::setICid(uint64_t ICid)
+/**
+ * @brief Function saves the mRegisteredClient attribute to persitant storage
+ *        To correctly persit an entry, the values must be stored in the structures attributes
+ *
+ * @return CHIP_ERROR
+ */
+CHIP_ERROR ClientMonitoringRegistrationTable::SaveToStorage()
 {
-    mRegisteredClient.ICid = ICid;
+    VerifyOrReturnError(mRegisteredClient.IsValid(), CHIP_ERROR_INVALID_ARGUMENT);
+
+    uint8_t buffer[kRegStorageSize] = { 0 };
+    TLV::TLVWriter writer;
+
+    writer.Init(buffer);
+    ReturnErrorOnFailure(mRegisteredClient.EncodeForWrite(writer, TLV::AnonymousTag()));
+    ReturnErrorOnFailure(writer.Finalize());
+
+    return mStorage.SyncSetKeyValue(DefaultStorageKeyAllocator::ClientMonitoringTableEntry(mRegisteredClient.fabricIndex).KeyName(),
+                                    buffer, static_cast<uint16_t>(writer.GetLengthWritten()));
 }
 
-void ClientMonitoringRegistrationTable::setFabricIndex(FabricIndex fabric)
+/**
+ * @brief Accessor function that returns the client registration entry that was loaded for a fabric from persistant storage.
+ * @see LoadFromStorage
+ *
+ * @return ClientMonitoringRegistrationTable::ClientRegistrationEntry&
+ */
+ClientMonitoringRegistrationTable::ClientRegistrationEntry & ClientMonitoringRegistrationTable::GetClientRegistrationEntry()
```

Comment:
Move to the header to allow for inlining?

Reply:
done